File: 2023/01/part1.ts
import { solveCalendar } from '../util/solveCalendar'
import { writeOutput } from '../util/writeOutput'

function solver(input: string[]) {
    const output: string[] = []

    const answer = input.reduce((sum, line) => {
        const numbers = line.split('').filter((char) => /[1-9]/.test(char))
        const first = numbers[0]
        const last = numbers[numbers.length - 1]

        output.push(first + last + ' - ' + line)
        return sum + Number(first + last)
    }, 0)

    writeOutput({ day: 1, part: 1, output })
    return answer
}

solveCalendar({
    day: 1,
    sampleNr: 1,
    sampleAnswer: 142,
    solver, // 54634 ⭐️
})

File: 2024/src/01/part1.py
from ..util.problem_solver import problem_solver


def solver(lines: list[str]) -> int:
    left_list: list[int] = []
    right_list: list[int] = []
    for line in lines:
        left, right = line.split()
        left_list.append(int(left))
        right_list.append(int(right))

    left_list.sort()
    right_list.sort()

    sum = 0
    for left, right in zip(left_list, right_list):
        sum += abs(left - right)
    return sum


if __name__ == "__main__":
    problem_solver(
        day=1,
        solver=solver,
        sample_answer=11,
    )
